### ISSUE TITLE
Matching service occasionally returns wrong room number for both users

### ISSUE THREAD

**Issue by Singa-pirate** [CS3219-AY2425S1/cs3219-ay2425s1-project-g06#37]
There are 2 different room numbers in MatchRecord
When a match is found, it occasionally return the wrong one that is not used in the session

**Comment by Singa-pirate** [comment]
Cause:
In matching service, the roomNumber and questionId are returned when match is found (`"matched"` event). This uses the data from the user who **starts matching later**.
```typescript
io.to(socketId).emit("matched", {
  matchedWith: existingMatch.userId,
  roomNumber,
  questionId: question?.questionId,
});
io.to(existingMatch.socketId).emit("matched", {
  matchedWith: userId,
  roomNumber,
  questionId: question?.questionId,
});
```
When match is confirmed (`""match_success"`) event, the roomNumber and questionId for the session is from the user who **accepts the match earlier**.
```typescript
await prisma.sessionHistory.create({
  data: {
    roomNumber: matchedRecord.roomNumber,
    questionId: matchedRecord.questionId ?? userRecord.questionId ?? 0,
    isOngoing: true,
    userOneId: userRecord.userId,
    userTwoId: matchedRecord.userId,
  },
});
```
When **these two users don't match**, the roomNumber used is wrong.

**Comment by Singa-pirate** [comment]
Closes with #40
